cobc/codegen.c

```diff
@@ -8108,6 +8108,33 @@ output_perform_until (struct cb_perform *p, cb_tree l)
 	cb_tree				next;
 
 	if (l == NULL) {
+#if 0 /* FIXME: add back as option, because not conforming to ISO */
+		if (CB_EXCEPTION_ENABLE (COB_EC_BOUND_SUBSCRIPT)) {
+			cb_tree	xn;
+			/* Check all INDEXED BY variables used in VARYING */
+			for (xn = p->varying; xn; xn = CB_CHAIN (xn)) {
+				struct cb_field		*q;
+				v = CB_PERFORM_VARYING (CB_VALUE (xn));
+				if (!v->name) continue;
+				f = CB_FIELD_PTR (v->name);
+				if (!f->flag_indexed_by) continue;
+				if (!f->index_qual) continue;
+				q = f->index_qual;
```

Comment:
we don't need `q` here (just reset `f`);
_maybe_ localize `f`'s definition here and in the debugging code below
and _maybe maybe_ you find a nice way to use less/no `continue` here - but "maybe"s are optional